File: app/src/main/java/x/chestnut/code/snippet/ui/fragment/lazyLoad/FragmentLazyLoadActivity.java
package x.chestnut.code.snippet.ui.fragment.lazyLoad;

import java.util.ArrayList;
import java.util.List;

import androidx.fragment.app.Fragment;
import androidx.viewpager.widget.ViewPager;
import x.chestnut.code.snippet.R;
import x.chestnut.code.snippet.base.BaseActivity;

public class FragmentLazyLoadActivity extends BaseActivity {

    @Override
    public int getLayoutId() {
        setTitle("FragmentLazyLoad");
        return R.layout.activity_fragment_lazy_load;
    }

    @Override
    public void lazyLoadViewAfterOnResume() {

        ViewPager viewPager;
        List<Fragment> list;

        list = new ArrayList<>();
        list.add(BlankFragment.newInstance("One"));
        list.add(BlankFragment.newInstance("Two"));
        list.add(BlankFragment.newInstance("Three"));

        MyViewPagerAdapter myAdapter = new MyViewPagerAdapter(getSupportFragmentManager(), list);
        viewPager = (ViewPager) findViewById(R.id.view_pager);
        viewPager.setAdapter(myAdapter);
        viewPager.setCurrentItem(0);
    }
}

File: app/src/main/java/x/chestnut/code/snippet/ui/fragment/lazyLoad/FragmentLazyLoadActivity.kt
package x.chestnut.code.snippet.ui.fragment.lazyLoad

import android.view.View
import androidx.fragment.app.Fragment
import androidx.viewpager.widget.ViewPager
import x.chestnut.code.snippet.R
import x.chestnut.code.snippet.base.BaseActivity
import x.chestnut.code.snippet.ui.fragment.lazyLoad.BlankFragment.Companion.newInstance
import java.util.*

class FragmentLazyLoadActivity : BaseActivity() {

    override val layoutId: Int
        get() {
            setTitle("FragmentLazyLoad")
            return R.layout.activity_fragment_lazy_load
        }

    override fun lazyLoadViewAfterOnResume() {
        val list: MutableList<Fragment>
        list = ArrayList()
        list.add(newInstance("One"))
        list.add(newInstance("Two"))
        list.add(newInstance("Three"))
        val myAdapter = MyViewPagerAdapter(supportFragmentManager, list)
        val viewPager: ViewPager? = findViewById<View>(R.id.view_pager) as? ViewPager
        viewPager?.adapter = myAdapter
        viewPager?.currentItem = 0
    }
}